omegaconf/_utils.py

```diff
@@ -496,7 +496,12 @@ def valid_value_annotation_type(type_: Any) -> bool:
 
 
 def _valid_dict_key_annotation_type(type_: Any) -> bool:
-    return type_ is None or issubclass(type_, str) or issubclass(type_, Enum)
+    return (
```

Comment:
Was this change needed to support the pickling DictConfig and ListConfig?

Reply:
Yes.